folly/compression/Compression.cpp

```diff
@@ -44,6 +44,9 @@
 #endif
 
 #if FOLLY_HAVE_LIBBZ2
+#if defined(_WIN32) && !defined(NOMINMAX)
+#define NOMINMAX 1
+#endif
 #include <bzlib.h>
```

Comment:
There are more things than just `NOMINMAX` that need to be handled with `Windows.h`, so just include `<folly/portability/Windows.h>` instead. It also means you don't have to guard it with `#ifdef _WIN32` anymore either :)